File: backend/src/main/java/com/intellidesk/knowledge/KnowledgeBase.java
package com.intellidesk.knowledge;

import com.baomidou.mybatisplus.annotation.*;
import lombok.Data;

import java.time.LocalDateTime;

@Data
@TableName("knowledge_base")
public class KnowledgeBase {

    @TableId(type = IdType.AUTO)
    private Long id;

    private Long workspaceId;
    private String name;
    private String description;
    private String chunkStrategy;
    private Integer chunkSize;
    private Integer chunkOverlap;
    private String status;
    private Long createdBy;

    @TableField(fill = FieldFill.INSERT)
    private LocalDateTime createdAt;

    @TableField(fill = FieldFill.INSERT_UPDATE)
    private LocalDateTime updatedAt;
}

File: frontend/src/types/knowledgeBase.ts
export type ChunkStrategy = 'FIXED_SIZE' | 'RECURSIVE';

export interface KnowledgeBase {
  id: number;
  workspaceId: number;
  name: string;
  description?: string | null;
  chunkStrategy: ChunkStrategy;
  chunkSize: number;
  chunkOverlap: number;
  status?: string;
  createdBy?: number;
  createdAt?: string;
  updatedAt?: string;
}

export interface KnowledgeBaseRequest {
  name: string;
  description?: string;
  chunkStrategy: ChunkStrategy;
  chunkSize: number;
  chunkOverlap: number;
}

export type KnowledgeBaseCreateRequest = KnowledgeBaseRequest;
export type KnowledgeBaseUpdateRequest = KnowledgeBaseRequest;

export interface PageResponse<T> {
  items: T[];
  page: number;
  size: number;
  total: number;
}
